lib/puppet/provider/vcsrepo.rb

```diff
@@ -16,7 +16,24 @@ def check_force
   def set_ownership
     owner = @resource.value(:owner) || nil
     group = @resource.value(:group) || nil
-    FileUtils.chown_R(owner, group, @resource.value(:path))
+    excludes = @resource.value(:excludes) || nil
+    if excludes.nil? || excludes.empty?
+      FileUtils.chown_R(owner, group, @resource.value(:path))
+    else
+      chown_r_with_excludes
+    end
+  end
+
+  def chown_r_with_excludes
+    owner = @resource.value(:owner) || nil
+    group = @resource.value(:group) || nil
+    FileUtils.chown(owner, group, files)
```

Comment:
since `owner` and `group` are available in `set_ownership` would it make more sense to move this to the `else` block and remove this function all together ?

Reply:
Right, sorry about that. I tried to Mock that function in a previous but rollback because it didn't add value.